Fixed Create Admin Issue and Added Logging

liaodrake opened the pull request:
Fixes the problem I experienced with Issue #60

jeffutter commented:
Thanks for your pull request. I'm ok with the log file, although I think I would prefer it be echoed to screen _and_ the log file (you can do this with 'tee').

As far as addUser vs createUser, i think this changed in some version of mongoDB. I'll see if I can track this down and get back to you soon.

jeffutter commented:
It appears that mongo 2.4 used addUser, while 2.6 uses createUser. This plugin attempts to support both. New installs should use 2.6, while old installs should use 2.4. I suppose my thinking was that if you are running the install script you would be using 2.6. If you are running into problems related to this, I suppose the install script should support both.

Can you see if you have jeffutter/mongodb:2.4 or jeffutter/mongodb:latest installed on your system? Can you also add logging to log the ```db_image``` variable? The vercomp function may be failing.

Again, thank you for your work. Please let me know if you have any questions.

liaodrake commented:
Hi, I have added `tee` to the logging functionality and modified what is being logged.

here is what my install is producing.
```
Plugin Install Dokku MongoDB

-- Build MongoDB Image
MongoDB Version: 2.6.3
db_image: jeffutter/mongodb
previous_mongo_install= false

-- Run Setup
id: a7a16110c1445518ed4d055a00184ae085e06ccbb60d5f58a6af3ba5e7230b79
mongodb_ip: 127.0.0.1
mongodb_port: 49157

-- Connecting to server and creating admin user

-- Try [createUser] to create admin
MongoDB shell version: 2.4.9
connecting to: 127.0.0.1:49157/admin
Thu Mar 26 19:41:48.535 TypeError: Property 'createUser' of object admin is not a function

-- Try [addUser] to create admin
Admin user created successfully

Plugin Install Dokku MongoDB Completed
```

I am fairly new to Dokku, so I am unsure where to check for `jeffutter/mongodb:2.4` or `jeffutter/mongodb:latest` being installed.

jeffutter commented:
OK. This looks good. I'm not sure I love trying both addUser and createUser but I'll take it until I have a more reliable way to determine what mongodb version we're dealing with.
